sphinx/testing/_internal/pytest_util.py

```diff
@@ -0,0 +1,415 @@
+"""Internal utility functions for interacting with pytest."""
+
+from __future__ import annotations
+
+__all__ = ()
+
+import os
+import warnings
+from contextlib import contextmanager
+from typing import TYPE_CHECKING, Literal, TypeVar, overload
+
+import pytest
+from _pytest.nodes import Node as PytestNode
+from _pytest.nodes import get_fslocation_from_item
+
+from sphinx.testing._internal.warnings import MarkWarning, NodeWarning, SphinxTestingWarning
+
+if TYPE_CHECKING:
+    from collections.abc import Callable, Collection, Generator, Iterable
+    from typing import Any, ClassVar, Final
+
+    T = TypeVar('T')
+    DT = TypeVar('DT')
+    NodeType = TypeVar('NodeType', bound="PytestNode")
+
+
+class TestRootFinder:
+    """Object responsible for finding the testroot files in *rootdir*.
+
+    For instance::
+
+        finder = TestRootFinder('/foo/bar', 'test-', 'default')
+
+    describes a testroot root directory at ``/foo/bar/roots``. The name of the
+    directories in ``/foo/bar/roots`` consist of a *prefix* and an *ID* (in
+    this case, the prefix is ``test-`` and the default *ID* is ``default``).
```

Comment:
```suggestion
        finder = TestRootFinder('/foo/bar', 'test-', 'default')

    describes a testroot root directory at ``/foo/bar/``. The name of the
    directories in ``/foo/bar/`` consist of a *prefix* and an *ID* (in this
    case, the prefix is ``test-`` and the default *ID* is ``default``).
```

Reply:
Good catch (it was something I did not touch since weeks ago).